File: src/programmers/Pg67256.java
package programmers;

public class Pg67256 {
}
class Solution67256 {
    public String solution(int[] numbers, String hand) {
        String answer = "";
        int nowLeft = 10;
        int nowRight = 12;
        for(int number : numbers){
            if(number ==0) number = 11;
            if(number%3 == 1) {
                answer+="L";
                nowLeft = number;
            }
            else if(number%3 == 0) {
                answer+="R";
                nowRight = number;
            }
            else{
                int numY = (number-1)/3;
                int numX = (number-1)%3;
                int leftLength = Math.abs(numY-(nowLeft-1)/3) + Math.abs(numX-(nowLeft-1)%3);
                int rightLength = Math.abs(numY-(nowRight-1)/3) + Math.abs(numX-(nowRight-1)%3);
                if(leftLength>rightLength){
                    answer+="R";
                    nowRight = number;
                }
                else if(leftLength<rightLength){
                    answer+="L";
                    nowLeft = number;
                }
                else{
                    if(hand.equals("left")){
                        answer+="L";
                        nowLeft = number;
                    }
                    else {
                        answer+="R";
                        nowRight = number;
                    }
                }
            }
        }
        return answer;
    }
}

File: src/programmers/python/lv1/Pg67256.py
# 키패드 누르기

def solution(numbers, hand):
    answer = ''
    now_L = [3,0]
    now_R = [3,2]
    for i in range(len(numbers)):
        if numbers[i] in [1,4,7]:
            answer+="L"
            now_L = [(numbers[i]-1)//3, (numbers[i]-1)%3]
        elif numbers[i] in [3,6,9]:
            answer+="R"
            now_R = [(numbers[i]-1)//3, (numbers[i]-1)%3]
        else:
            a,now_L,now_R = distance(numbers[i],hand,now_L,now_R)
            answer+=a
    return answer

def distance(number, hand, now_L, now_R):
    num = number
    if number == 0:
        num = 11
    dis_L = abs((num-1)//3 - now_L[0]) + abs((num-1)%3 - now_L[1])
    dis_R = abs((num-1)//3 - now_R[0]) + abs((num-1)%3 - now_R[1])
    next_pos =[(num-1)//3, (num-1)%3]
    if dis_L < dis_R or (dis_L == dis_R and hand=="left"):
        return "L", next_pos, now_R
    elif dis_L > dis_R or (dis_L == dis_R and hand=="right"):
        return "R", now_L, next_pos